usecase/csv/csv.go

```diff
@@ -0,0 +1,49 @@
+package csv
+
+import (
+	"errors"
+	"net/http"
+	"strconv"
+	"bootcamp/domain/model"
+	"bootcamp/utils"
+	"github.com/gorilla/mux"
+)
+
+/*
+GetPokemon returns a Pokemon list read from a CSV file
+If query params exists or /{id}, filter the Pokemon
+Otherwise, returns the list
+*/
+func GetPokemon(r *http.Request) (model.PokemonList, error) {
+	pokemonList, err := utils.ReadCSV()
```

Comment:
You should handle the error right after you receive it instead of doing more things before that. I'm guessing that if there was an error reading the csv, then the whole flow should fail and you should return here

Reply:
*